tests/test_examples.py

```diff
@@ -1294,3 +1297,59 @@ def test_second_stage_optimization():
     np.testing.assert_allclose(field[0].R0, 3.5)  # this value was fixed
     np.testing.assert_allclose(field[0].B0, 1)  # toroidal field (no change)
     np.testing.assert_allclose(field[1].B0, 0, atol=1e-12)  # vertical field (vanishes)
+
+
+@pytest.mark.unit
+def test_coilset_geometry_optimization():
```

Comment:
This is supposed to test the two new objectives in a simple second-stage tokamak coil optimization, but it could use improvement. 

The initial coil set is shown below: 
![image](https://github.com/PlasmaControl/DESC/assets/33005725/53f43d3c-588f-44a5-b591-cf5471d05cb1)

The idea is to reduce the radius of the coils (to match the target plasma-coil distance), and to move the coils so they are evenly spaced around the torus (to maximize coil-coil distance). But this is the optimized result: 
![image](https://github.com/PlasmaControl/DESC/assets/33005725/e3a67d42-26d4-4ba9-9c86-4b56b669fc09)

It would be ideal for this test if we could constrain the coils to remain in a constant $\phi$ plane. Could we add an option to `FourierPlanarCurve` to parameterize its `center` and `normal` in $(R,\phi,Z)$ coordinates instead of $(X,Y,Z)$ coordinates? Then that constraint would be easy to enforce.